src/translate.py

```diff
@@ -3781,7 +3781,7 @@ def read_pointer() -> Optional[Expression]:
 
         def for_element_type(type: Type) -> Optional[str]:
             """Return the initializer for a single element of type `type`"""
-            if type.is_int() or type.is_float():
+            if type.is_int() or type.is_likely_float():
```

Comment:
Should this be `is_likely_int()`? 
Or even `is_numeric()`? I think we actually care that `kind` is a subset of `INT | FLOAT` -- not that it is likely one or the other? We can leave it up to `Literal.format()` to decide how to format the number?

Reply:
I think `kind == (K_INT | K_FLOAT)` won't actually ever happen in practice... we can only get there by starting from K_ANYREG or K_ANY, and anything that removes K_PTR will also remove K_FLOAT. Similarly is_likely_int() practically implies is_int(). Not sure what that implies the best option here is

Reply:
oh actually what should be done is to combine int and pointer handling; then we can use a single code path and leave things to Literal.format